### PR TITLE
INTLY-3820 Backport changes to providers

### PR DESCRIPTION
**Pull request by ciaranRoche**
Co-authored-by: dimitraz <dzuccare@redhat.com>

## Overview

Jira: https://issues.jboss.org/browse/INTLY-3820

## Verify
- Run `make cluster/clean`
- Run `make cluster/prepare`
- Set `postgres` and `redis` CR's to `workshop` 
- Seed the cluster `make cluster/seed/postgres` `make cluster/seed/redis`
- Run `make run`
- Verify both `postgres` and `redis` have created successfully in cluster
- Remove both `postgres` and `redis` CR's 
- Verify both have deleted successfully
- Stop operator
- Set `postgres` and `redis` CR's to `managed`
- Seed the cluster `make cluster/seed/postgres` `make cluster/seed/redis` `make cluster/seed/blobstorage` `make cluster/seed/smtp`
- Grab a cup of coffee and wait for everything to provision
- Verify everything works as expected
- Remove all CR's 
- Verify everything deletes as expected

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,7 +65,7 @@ cluster/clean:
 .PHONY: test/unit
 test/unit:
 	@echo Running tests:
-	gotest -v -covermode=count -coverprofile=coverage.out ./pkg/...
```
Comment (austincunningham) [inline Makefile]: Can you add this back again :)

File: deploy/crds/integreatly_v1alpha1_postgres_cr.yaml [hunk 2]
```diff
@@ -11,4 +11,4 @@ spec:
   # i want a postgres storage of a development-level tier
   tier: development
   # i want a postgres storage for the type managed
-  type: managed
+  type: workshop
```
Comment (austincunningham) [inline deploy/crds/integreatly_v1alpha1_postgres_cr.yaml]: Will we leave it at managed ?

File: deploy/crds/integreatly_v1alpha1_redis_cr.yaml [hunk 3]
```diff
@@ -11,4 +11,4 @@ spec:
   # i want a redis storage of a development-level tier
   tier: development
   # i want a redis storage for the type managed
-  type: managed
```
Comment (austincunningham) [inline deploy/crds/integreatly_v1alpha1_redis_cr.yaml]: ^^ :eyes:

File: pkg/resources/config.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import (
+	"os"
+	"strconv"
+	"time"
+)
+
+const DefaulReconcileTime = 300
+
+// returns envar for reconcile time else returns default time
+func GetReconcileTime() time.Duration {
```
Comment (aidenkeating) [inline pkg/resources/config.go]: We should probably have a test for this, it's an important piece of logic in the operator overall, we don't want any regressions or unexpected behaviour here

File: doc/postgresql.md [hunk 5]
```diff
@@ -6,7 +6,8 @@ To seed a Kubernetes/Openshift cluster with an example Postgres resource:
 $ make cluster/prepare 
 $ make cluster/seed/postgres
 ```
-
+### AWS Strategy
+A JSON object containing two keys: `region`, which is the [AWS region code](https://docs.aws.amazon.com/general/latest/gr/rande.html#ses_region), a `createStrategy`, which is a JSON representation of [this struct](https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#CreateDBInstanceInput), and a deleteStrategy, which is a JSON representation of [this struct](https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#DeleteDBInstanceInput)
```
Comment (dimitraz) [inline doc/postgresql.md]:
```suggestion
A JSON object containing three keys: `region`, which is the [AWS region code](https://docs.aws.amazon.com/general/latest/gr/rande.html#ses_region), a `createStrategy`, which is a JSON representation of [this struct](https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#CreateDBInstanceInput), and a `deleteStrategy`, which is a JSON representation of [this struct](https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#DeleteDBInstanceInput).
```

File: deploy/crds/integreatly_v1alpha1_redis_cr.yaml (hunk 3 above)
Comment (dimitraz) [reply deploy/crds/integreatly_v1alpha1_redis_cr.yaml]: Can we also make the `secretRef` names the same across all CRs?

File: pkg/controller/smtpcredentialset/smtpcredentialset_controller.go [hunk 6]
```diff
@@ -161,8 +156,8 @@ func (r *ReconcileSMTPCredentialSet) Reconcile(request reconcile.Request) (recon
 			return nil
 		})
 		if err != nil {
-			if err = resources.UpdatePhase(r.ctx, r.client, instance, v1alpha1.PhaseFailed, "failed to reconcile instance secret"); err != nil {
-				return reconcile.Result{}, err
+			if updateErr := resources.UpdatePhase(r.ctx, r.client, instance, v1alpha1.PhaseFailed, "failed to reconcile instance secret"); updateErr != nil {
+				return reconcile.Result{}, updateErr
 			}
 			return reconcile.Result{}, errorUtil.Wrapf(err, "failed to reconcile blob storage instance secret %s", sec.Name)
```
Comment (dimitraz) [inline pkg/controller/smtpcredentialset/smtpcredentialset_controller.go]:
```suggestion
			return reconcile.Result{}, errorUtil.Wrapf(err, "failed to reconcile smtp credential set instance secret %s", sec.Name)
```

File: pkg/resources/config.go (hunk 4 above)
Comment (ciaranRoche) [reply pkg/resources/config.go]: Will catch in a follow up PR

File: Makefile (hunk 1 above)
Comment (ciaranRoche) [reply Makefile]: We opted to go back to `go test` to avoid adding more dependencies to the repo

**Pull request merged**